queries/futures/utils.ts

```diff
@@ -290,92 +294,83 @@ export const mapMarginTransfers = (
-			?.map(
-				({
-					id,
-					lastTxHash,
-					timestamp,
-					market,
-					asset,
-					account,
-					isOpen,
-					isLiquidated,
-					size,
-					feesPaid,
-					netFunding,
-					netTransfers,
-					totalDeposits,
-					initialMargin,
-					margin,
-					entryPrice,
-					exitPrice,
-					pnl,
-					pnlWithFeesPaid,
-					openTimestamp,
-					closeTimestamp,
-					totalVolume,
-					trades,
-					avgEntryPrice,
-				}: RawPosition) => {
-					const entryPriceWei = new Wei(entryPrice, 18, true);
-					const exitPriceWei = new Wei(exitPrice || 0, 18, true);
-					const sizeWei = new Wei(size, 18, true);
-					const feesWei = new Wei(feesPaid || 0, 18, true);
-					const netFundingWei = new Wei(netFunding || 0, 18, true);
-					const netTransfersWei = new Wei(netTransfers || 0, 18, true);
-					const totalDepositsWei = new Wei(totalDeposits || 0, 18, true);
-					const initialMarginWei = new Wei(initialMargin, 18, true);
-					const marginWei = new Wei(margin, 18, true);
-					const pnlWei = new Wei(pnl, 18, true);
-					const pnlWithFeesPaidWei = new Wei(pnlWithFeesPaid, 18, true);
-					const totalVolumeWei = new Wei(totalVolume, 18, true);
-					const avgEntryPriceWei = new Wei(avgEntryPrice, 18, true);
-					return {
-						id: Number(id.split('-')[1].toString()),
-						transactionHash: lastTxHash,
-						timestamp: timestamp * 1000,
-						openTimestamp: openTimestamp * 1000,
-						closeTimestamp: closeTimestamp * 1000,
-						market,
-						asset: utils.parseBytes32String(asset) as FuturesMarketAsset,
-						account,
-						isOpen,
-						isLiquidated,
-						size: sizeWei.abs(),
-						feesPaid: feesWei,
-						netFunding: netFundingWei,
-						netTransfers: netTransfersWei,
-						totalDeposits: totalDepositsWei,
-						initialMargin: initialMarginWei,
-						margin: marginWei,
-						entryPrice: entryPriceWei,
-						exitPrice: exitPriceWei,
-						pnl: pnlWei,
-						pnlWithFeesPaid: pnlWithFeesPaidWei,
-						totalVolume: totalVolumeWei,
-						trades,
-						avgEntryPrice: avgEntryPriceWei,
-						leverage: marginWei.eq(wei(0))
-							? wei(0)
-							: sizeWei.mul(entryPriceWei).div(marginWei).abs(),
-						side: sizeWei.gte(wei(0)) ? PositionSide.LONG : PositionSide.SHORT,
-					};
-				}
-			)
-			.filter(({ isOpen }: { isOpen: boolean }) => {
-				if (openOnly) {
-					return isOpen;
-				} else {
-					return true;
-				}
-			})
-			.filter(({ id }: { id: number }) => id !== 0) ?? null
+	return futuresPositions.map(
+		({
+			id,
+			lastTxHash,
+			openTimestamp,
+			closeTimestamp,
+			timestamp,
+			market,
+			asset,
+			account,
+			abstractAccount,
+			accountType,
+			isOpen,
+			isLiquidated,
+			trades,
+			totalVolume,
+			size,
+			initialMargin,
+			margin,
+			pnl,
+			feesPaid,
+			netFunding,
+			pnlWithFeesPaid,
+			netTransfers,
+			totalDeposits,
+			entryPrice,
+			avgEntryPrice,
+			exitPrice,
+		}: FuturesPositionResult) => {
+			const entryPriceWei = weiFromWei(entryPrice).div(ETH_UNIT);
```

Comment:
How come we need to divide by ETH in these conversions? `weiFromWei` is a util for converting a BigNum wei to Synthetix wei, e.g. so BigNumber('1000000000000000000') gets set as wei('1')

Reply:
Looks like this is because the original value is a `Wei` value with a precision of 0. So the function just returns the same value again, with a new precision of 18.

To get the "real" value we need to divide here, do you think that means we should update the `weiFromWei` function to handle this when precision values are different?

Reply:
I just tried here changing this to only `weiFromWei(entryPrice)` and it seemed to work as expected. The `weiFromWei` func is really just a helper for the `new Wei(entryPrice, 18, true)` calls that were replaced. Unless there's some other change that impacts these as I was only trying on dev.

Reply:
That's not what I'm seeing. When I remove the divisor I get this number value for the avg entry price: `2.0122782e+22`
<img width="931" alt="image" src="https://user-images.githubusercontent.com/10401554/194120532-b66dea13-b868-42c0-bcd2-6d631901f528.png">

Reply:
This function _does_ work for orders, where the values returned from the query are a string. This query returns `Wei` values with a precision of 0 instead of 18, so they are simply converted to `Wei` with a precision of 18 without doing the conversion.

Reply:
I just added an if statement to that function to check if the input is already a `Wei` value. If it is, we convert from the given precision to 18.